src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,792 @@
+            {
+                if (_semanticModel.GetSymbolInfo(invocationExpression, _cancellationToken).Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 0 && methodSymbol.ReturnType?.SpecialType == SpecialType.System_Int32)
+                {
+                    // before var count = (from a in b select a).Count();
+                    // after
+                    // var count = 0;
+                    // foreach (var a in b)
+                    // {
+                    //     count++;
+                    //  }
+                    return TryConvertIfInInvocation(
+                            source,
+                            invocationExpression,
+                            (variableIdentifier, expression) => SyntaxFactory.ExpressionStatement(
+                                SyntaxFactory.PostfixUnaryExpression(SyntaxKind.PostIncrementExpression, variableIdentifier)), // Generating 'count++'
+                            SyntaxFactory.LiteralExpression(SyntaxKind.NumericLiteralExpression, SyntaxFactory.Literal(0)), // count = 0
+                            variableName: "count",
+                            out documentUpdateInfo);
+                }
+
+                documentUpdateInfo = default;
+                return false;
+            }
+
+            private bool TryConvertIfInToListInvocation(
+                QueryExpressionSyntax source,
+                InvocationExpressionSyntax invocationExpression,
+                out DocumentUpdateInfo documentUpdateInfo)
+            {
+                // before var list = (from a in b select a).ToList();
+                // after
+                // var list = new List<T>();
+                // foreach (var a in b)
+                // {
+                //     list.Add(a)
+                // }
+                if (_semanticModel.GetSymbolInfo(invocationExpression, _cancellationToken).Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.Parameters.Length == 0)
+                {
+                    return TryConvertIfInInvocation(
+                              source,
+                              invocationExpression,
+                              (listIdentifier, expression) => SyntaxFactory.ExpressionStatement(SyntaxFactory.InvocationExpression(
+                                    SyntaxFactory.MemberAccessExpression(
+                                        SyntaxKind.SimpleMemberAccessExpression,
+                                        listIdentifier,
+                                        SyntaxFactory.IdentifierName(nameof(IList.Add))),
+                                    SyntaxFactory.ArgumentList(SyntaxFactory.SingletonSeparatedList(SyntaxFactory.Argument(expression))))),
+                              SyntaxFactory.ObjectCreationExpression(
+                                  methodSymbol.GenerateReturnTypeSyntax().WithAdditionalAnnotations(Simplifier.Annotation),
```

Comment:
>GenerateReturnTypeSyntax [](start = 47, length = 24)

Is this going to take care of adding namespaces?

Reply:
yes, and then we reduce them with the simplifier

---
In reply to: [176929095](https://github.com/dotnet/roslyn/pull/25362#discussion_r176929095) [](ancestors = 176929095)